src/modules/Dashboard/features/Managers/Popups/SelectRespondentsPopup/SelectRespondentsPopup.test.tsx

```diff
@@ -0,0 +1,164 @@
+    },
+    applet: mockedApplet,
+    workspacesRoles: initialStateData,
+  },
+  users: {
+    allRespondents: {
+      ...initialStateData,
+      data: {
+        result: [mockedRespondent, mockedRespondent2],
+        count: 2,
+      },
+    },
+  },
+};
+
+const mockedCloseFn = jest.fn();
+const getPopup = (withSelectedRespondents = true) => (
+  <SelectRespondentsPopup
+    appletName="testName"
+    appletId={mockedAppletId}
+    user={mockedManager}
+    selectedRespondents={withSelectedRespondents ? [mockedRespondentId] : []}
+    selectRespondentsPopupVisible={true}
+    onClose={mockedCloseFn}
+  />
+);
+
+const successfulResponse = {
+  status: ApiResponseCodes.SuccessfulResponse,
+  data: {
+    result: [mockedRespondent, mockedRespondent2],
+    count: 2,
+  },
+};
+
+describe('SelectRespondentsPopup component tests', () => {
+  test('should appear popup without selected respondents', async () => {
+    mockAxios.get.mockResolvedValueOnce(successfulResponse);
+    renderWithProviders(getPopup(false), { preloadedState });
+    const tableRows = screen.queryAllByTestId('table-row');
+    const select = screen.getByTestId('select-respondents-popup-search-across');
+    const selectInput = await waitFor(() => select.querySelector('input'));
+
+    await waitFor(() => {
+      expect(selectInput?.value).toBe(SearchAcross.All);
+      expect(tableRows.length).toBe(2);
+      expect(screen.getByTestId('dashboard-managers-select-respondents-popup')).toBeInTheDocument();
+      expect(screen.getByText('Mocked Respondent')).toBeInTheDocument();
+      expect(screen.getByText('Test Respondent')).toBeInTheDocument();
+      expect(
+        screen.getByText('To proceed, at least 1 Respondent should be selected.'),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('should appear popup with selected respondents', async () => {
+    mockAxios.get.mockResolvedValueOnce(successfulResponse);
+    renderWithProviders(getPopup(), { preloadedState });
+
+    await waitFor(() => {
+      expect(screen.getByText('1 Respondent selected')).toBeInTheDocument();
+    });
+  });
+
+  test('should confirm popup with selected respondents', async () => {
+    mockAxios.get.mockResolvedValueOnce(successfulResponse);
+    renderWithProviders(getPopup(), { preloadedState });
+
+    fireEvent.click(screen.getByText('Confirm'));
+
+    await waitFor(() => {
+      expect(mockedCloseFn).toBeCalledWith([mockedRespondentId]);
+    });
+  });
+
+  describe('should search respondents', () => {
+    test('across all', async () => {
+      mockAxios.get.mockResolvedValueOnce(successfulResponse);
+      renderWithProviders(getPopup(), { preloadedState });
+      const mockedSearchValue = 'mockedSearchValue';
+
+      const search = await waitFor(() => screen.getByTestId('dashboard-select-respondents-search'));
+      const searchInput = search.querySelector('input');
+      searchInput && fireEvent.change(searchInput, { target: { value: mockedSearchValue } });
```

Comment:
why did you check if `searchInput` is not valid?

Reply:
because with querySelector it could be null